scripts/bake-ide-ami.ps1

```
@@ -468,9 +468,53 @@ $jsonObject = @"
                 $ChefRecipe = "VLWebServer::IDEBaseAzure"
             }
 
+            # 163204: Generate PFX using secret from Azure Vault
             # Make sure the session is initialised correctly
             ReConnect-Session
 
+            # set the license key password
+            $script:licensekeypassword = $Script:password;
+    
+            # ensures that the PFX Certificates are present in the license key path
+            $vmPfxCertificates = @{IntegratorLicensePrivateKey = "LANSAIntegratorLicense"; ScalableLicensePrivateKey = "LANSAScalableLicense"};
+            foreach ($certificateName in $vmPfxCertificates.Keys) {
+                $secret = Get-AzKeyVaultSecret -VaultName $KeyVault -Name $certificateName
+                if ( $secret ) {
+                    # Write to a file
+                    Write-Verbose "$(Log-Date) Create a PFX File for $certificateName Certificate"
+
+                    # contains the JSON object with data, password and data type
+                    $certData = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secret.SecretValueText))| ConvertFrom-Json
+    
+                    $pfx = New-Object Security.Cryptography.X509Certificates.X509Certificate2
+                    $pfx.Import([System.Convert]::FromBase64String($certData.data), $certData.password, [Security.Cryptography.X509Certificates.X509KeyStorageFlags]::Exportable + [Security.Cryptography.X509Certificates.X509KeyStorageFlags]::PersistKeySet)
+                    $pfxProtectedBytes = $pfx.Export([Security.Cryptography.X509Certificates.X509ContentType]::Pkcs12, $Script:password)
+                    [System.IO.File]::WriteAllBytes("$Script:LicenseKeyPath\$($vmPfxCertificates[$certificateName]).pfx", $pfxProtectedBytes)
+                } else {
+                    Write-Verbose "$(Log-Date) Create $certificateName Certificate"    
```

If a certificate is not in the Azure Key Vault then it is a fatal error. The certificate that is in Key Vault MUST be used.

So the `else` statement is incorrect, it should throw a fatal error
```
                    } else {...
```

---

Hi @robe070 
Thank you for your comments, agreed. I had added the throw statement. Please review and let me know your inputs.